numba/core/typing/new_builtins.py

```diff
@@ -427,14 +390,11 @@ def impl(self, other):
 @overload_method(types.PythonFloat, "__add__")
 @overload_method(types.PythonFloat, "__radd__")
 def py_float__add__(self, other):
-    # evaluate type
     def impl(self, other):
-        if hasattr(other, "__float__"):
-            asfloat = other.__float__()
-            if asfloat is NotImplemented:
-                return NotImplemented
-            else:
-                return float_add_float(self, asfloat)
+        if isinstance(other, float):
+            return float_add_float(self, other)
+        elif isinstance(other, (int, bool, np.float64)):
```

Comment:
Python float should not know about numpy types. Handling of `np.float64` should be done in NumPy binops. Also, the result type would be numpy types.

Reply:
The primary issue is if we remove this the following behaviour happens:

```
import numba
import numpy as np

@numba.njit
def foo(v, w):
    return v.__add__(w)

x = 1.1
y = np.float64(5.5)
print(foo.py_func(x, y))
# 6.6
print(foo(x, y))
# NotImplemented
```